### PR TITLE
r/opsworks_ecs_cluster_layer - add new resource

### LINKED ISSUE #182: provider/aws: Support OpsWorks ECS Layers
_This issue was originally opened by @eedwardsdisco as hashicorp/terraform#7036. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Request for support of OpsWorks ECS Layer Type, including the ability to associate an ECS Cluster with an OpsWorks ECS Layer.

http://docs.aws.amazon.com/opsworks/latest/userguide/workinglayers-ecscluster.html

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #182 
Replaces #3024

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

Output from acceptance testing:

```
$ make testacc TESTS=TestAccOpsWorksEcsClusterLayer_ PKG=opsworks
--- PASS: TestAccOpsWorksEcsClusterLayer_basic (53.53s)
--- PASS: TestAccOpsWorksEcsClusterLayer_tags (117.74s)

--- PASS: TestAccOpsWorksMySQLLayer_basic (56.25s) -- to check the added validate attribute didn't break anything
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Validated acceptance tests:

```console
$ make testacc TEST=./aws/ TESTARGS='-run=estAccAWSOpsworksECSLayer_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws/ -v -count 1 -parallel 20 -run=estAccAWSOpsworksECSLayer_ -timeout 120m
=== RUN   TestAccAWSOpsworksECSLayer_basic
=== PAUSE TestAccAWSOpsworksECSLayer_basic
=== RUN   TestAccAWSOpsworksECSLayer_tags
=== PAUSE TestAccAWSOpsworksECSLayer_tags
=== CONT  TestAccAWSOpsworksECSLayer_basic
=== CONT  TestAccAWSOpsworksECSLayer_tags
--- PASS: TestAccAWSOpsworksECSLayer_basic (49.27s)
--- PASS: TestAccAWSOpsworksECSLayer_tags (119.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	119.047s

```

**Comment by darkpssngr** [comment]
If everything is fine can we merge this? I would like to use the upstream binary :)

File: website/docs/r/opsworks_ecs_cluster_layer.html.markdown [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+# Resource: aws_opsworks_ecs_cluster_layer
+
+Provides an OpsWorks ECS Cluster layer resource.
+
+## Example Usage
+
+```hcl
+resource "aws_opsworks_ecs_cluster_layer" "lb" {
+  stack_id       = aws_opsworks_stack.main.id
+  stats_password = "foobarbaz"
```
Comment (darkpssngr) [inline website/docs/r/opsworks_ecs_cluster_layer.html.markdown:18]: shouldn't this be `ecs_cluster_arn`?

Comment (darkpssngr) [inline website/docs/r/opsworks_ecs_cluster_layer.html.markdown:18]: @DrFaust92 shouldn't this be `ecs_cluster_arn` ?

Comment (DrFaust92) [reply website/docs/r/opsworks_ecs_cluster_layer.html.markdown]: fixed

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccOpsWorksEcsClusterLayer_\|TestAccOpsWorksMySQLLayer_basic\|TestAccOpsWorksPHPAppLayer_basic' PKG=opsworks
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/opsworks/... -v -count 1 -parallel 20  -run=TestAccOpsWorksEcsClusterLayer_\|TestAccOpsWorksMySQLLayer_basic\|TestAccOpsWorksPHPAppLayer_basic -timeout 180m
=== RUN   TestAccOpsWorksEcsClusterLayer_basic
=== PAUSE TestAccOpsWorksEcsClusterLayer_basic
=== RUN   TestAccOpsWorksEcsClusterLayer_tags
=== PAUSE TestAccOpsWorksEcsClusterLayer_tags
=== RUN   TestAccOpsWorksMySQLLayer_basic
=== PAUSE TestAccOpsWorksMySQLLayer_basic
=== RUN   TestAccOpsWorksPHPAppLayer_basic
=== PAUSE TestAccOpsWorksPHPAppLayer_basic
=== CONT  TestAccOpsWorksEcsClusterLayer_basic
=== CONT  TestAccOpsWorksMySQLLayer_basic
=== CONT  TestAccOpsWorksEcsClusterLayer_tags
=== CONT  TestAccOpsWorksPHPAppLayer_basic
--- PASS: TestAccOpsWorksMySQLLayer_basic (48.35s)
--- PASS: TestAccOpsWorksEcsClusterLayer_basic (57.83s)
--- PASS: TestAccOpsWorksPHPAppLayer_basic (60.54s)
--- PASS: TestAccOpsWorksEcsClusterLayer_tags (123.33s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/opsworks	126.958s
```

**Comment by ewbankkit** [comment]
@DrFaust92 Thanks for the contribution :tada: :clap:.

**Pull request merged**